utils/pubsub/pubsubClient.ts

```diff
@@ -172,4 +173,67 @@ export async function pollBuildStatus(projectId: string, location: string, build
     }
 	console.error('[pollBuildStatus] Build status check exceeded maximum attempts');
     return 'TIMEOUT'; // Indicate a timeout in build status check
+}
+
+export async function triggerCloudPatBuildUsingGcloudApi(user_id: string, topic_name: string, encrypted_github_pat: string, provider: string): Promise<CloudBuildStatus> {
```

Comment:
This and the other function can be written as one function with the number and type of variables different for the two cases